### PR TITLE
Handbook: Detail the Gutenberg release post process

### PR DESCRIPTION
**Pull request by priethor**
## Description
Updates the [Gutenberg release process handbook page](https://developer.wordpress.org/block-editor/contributors/code/release/#writing-the-release-notes-and-post) to expand the "Writing the Release Notes and Post" based on the last release iterations.

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
It looks good to me. Thank you for adding more details to how the process works in practice 👍🏻 

I think @youknowriad is the best person to perform another review.

File: docs/contributors/code/release.md [hunk 1]
```diff
@@ -50,22 +50,44 @@ Once released, all that's left to do is writing a release post on [make.wordpres
...
+
+#### 4. Running the performance tests
+
+Your post should also include a performance audit at the end, comparing the current Gutenberg release with both the previous one and the latest WordPress major version. You can use `bin/plugin/cli.js perf` to automatically do this for you by passing as parameters the tags to run the performance suite against, such as `bin/plugin/cli.js perf release/x.y release/x.z wp/a.b`
+
+#### 5. Publishing the post
+
+Compile this to a draft post on [make.wordpress.org/core](https://make.wordpress.org/core/); this post should be published after the actual release. Remember asking for peer review is encouraged by the [make/core posting guidelines](https://make.wordpress.org/core/handbook/best-practices/post-comment-guidelines/#peer-review)!
```
Comment (ntsekouras) [inline docs/contributors/code/release.md:90]: Now that we also have volunteers from the community, there is a great chance that they don't have access to `make.wordpress.org/core` to create a draft and publish. 

Should we direct them to the process of seeking access through `#meta` or maybe it's better to prepare this and ping someone from the core team?

**Review by ntsekouras — APPROVED** [review]
Always great to see such enhancements! Thanks @priethor 💯

File: docs/contributors/code/release.md [hunk 2]
```diff
@@ -50,22 +50,44 @@ Once released, all that's left to do is writing a release post on [make.wordpres
...
+
+These visual assets should maintain consistency with previous release posts; using lean, white themes helps in this regard and visually integrate well with the [make.wordpress.org/core](https://make.wordpress.org/core/) blog aesthetics. Including copyrighted material should be avoided, and browser plugins that can be seen in the browser canvas (spell checkers, form fillers, etc.) disabled when capturing the assets. If you would like help creating the visual assets, please ping somebody in the Gutenberg Core team in advance so that it can be planned for.
+
+#### 4. Running the performance tests
+
+Your post should also include a performance audit at the end, comparing the current Gutenberg release with both the previous one and the latest WordPress major version. You can use `bin/plugin/cli.js perf` to automatically do this for you by passing as parameters the tags to run the performance suite against, such as `bin/plugin/cli.js perf release/x.y release/x.z wp/a.b`
```
Comment (youknowriad) [inline docs/contributors/code/release.md:86]: I think now we have a job that is run on CI that gives you the results, so there's no need to run this on your own in theory.

**Review by youknowriad — APPROVED** [review]
Nice work here

File: docs/contributors/code/release.md (hunk 2 above)
Comment (priethor) [reply docs/contributors/code/release.md:86]: As far as I see there is such a job, although it failed for the 10.7.0 to 10.7.4 releases [when running the perf tests on the 5.7 branch](https://github.com/WordPress/gutenberg/runs/2676154517?check_suite_focus=true#step:7:151).

Comment (youknowriad) [reply docs/contributors/code/release.md:86]: I think @mcsf updated the job to use the latest WP release as base branch which should in theory make it more stable.

Comment (priethor) [reply docs/contributors/code/release.md:86]: Indeed, I failed to add that it should be fixed in 10.8 thanks to #32244 👍

**Pull request merged**